models/models_methods.go

```diff
@@ -559,8 +559,12 @@ func validatePipelines(config *BitriseDataModel) ([]string, error) {
 
 		if hasStages && hasWorkflows {
 			return pipelineWarnings, fmt.Errorf("pipeline (%s) has both stages and workflows", pipelineID)
-		} else if !hasStages && !hasWorkflows {
-			return pipelineWarnings, fmt.Errorf("pipeline (%s) should have at least 1 stage or workflow", pipelineID)
+		}
+
+		// A pipeline is considered valid if it has neither stages nor workflows.
+		// This is useful for the WFE to be able to save a pipeline that is not yet fully defined.
+		if !hasStages && !hasWorkflows {
+			continue
```

Comment:
Did you consider adding a warning in this case?

Reply:
We discussed it and then I forgot to add it. Thanks for the reminder.